### PR TITLE
Ref ARRUS-111: Extending the B-Mode display functionality

### PR DESCRIPTION
**Pull request by PiotrKarwat**
New features:

- cineloop
- persistence
- TGC on image data (manual & auto version)

Bug fixes:

- colorscale in colorbar now reflects the dynamic range of the B-Mode image

### CODE REVIEW

File: api/matlab/arrus/DuplexDisplay.m [hunk 1]
```diff
@@ -95,9 +161,29 @@ function updateImg(obj, data)
...
                 [nZPix,nXPix,~,~] = size(data);
                 
-                bmode = data(:,:,:,1);
-                bmode(isnan(bmode)) = -inf;
+                % update cineLoop
+                obj.cineLoopIndex = mod(obj.cineLoopIndex, obj.cineLoopLength) + 1;
+                obj.cineLoop(:,:,obj.cineLoopIndex,:) = data;
+                
+                % persistence
+                index = mod(obj.cineLoopIndex-(0:(numel(obj.persistence)-1))-1,obj.cineLoopLength)+1;
```
Comment (zkLog) [inline api/matlab/arrus/DuplexDisplay.m:169]: spacje wokół operatorów - +

File: api/matlab/arrus/DuplexDisplay.m [hunk 2]
```diff
@@ -12,26 +12,82 @@
...
             end
             
-            if nargin < 2
-                dynamicRange = [0 60];
-            else 
-                if ~all(size(dynamicRange) == [1 2])
-                    error("ARRUS:IllegalArgument", ...
-                        "Invalid dimensions of dynamic range vector, should be: [min max]")
-                end
+            if ~isscalar(powerThreshold) || ~isnumeric(powerThreshold)
```
Comment (zkLog) [inline api/matlab/arrus/DuplexDisplay.m:60]: Consider moving validation functions into addParameter() method.

File: api/matlab/arrus/DuplexDisplay.m (hunk 1 above)
Comment (PiotrKarwat) [reply api/matlab/arrus/DuplexDisplay.m:169]: Done

File: api/matlab/arrus/DuplexDisplay.m (hunk 2 above)
Comment (PiotrKarwat) [reply api/matlab/arrus/DuplexDisplay.m:60]: In future development - yes, for now it can stay this way.

Comment (PiotrKarwat) [reply api/matlab/arrus/DuplexDisplay.m]: Done

**Review by zkLog — APPROVED** [review]

**Review by zkLog — APPROVED** [review]

**Pull request merged**